### PR TITLE
BAH-1256 Add a button to start ad-hoc tele-consultation in the header section in the ‘Patient Dashboard’ Screen.

### PR DESCRIPTION
**Pull request by vishalreddysarasani**
Refer https://bahmni.atlassian.net/browse/BAH-1256

Added a button and info pop up box for ad-hoc teleconsultation changes
Screenshot from 2021-10-04 12-18-28

### CODE REVIEW

**Review by gsluthra — CHANGES_REQUESTED** [review]

File: ui/app/clinical/dashboard/views/clinicalDashboardHeader.html [hunk 1]
```diff
@@ -82,6 +82,10 @@
                 </ul>
             </div>
             <div class="dashboard-header-right-wrapper">
+                <div ng-if="::clinicalDashboardConfig.allowAdhocTeleConsultation" ng-click="startAdhocTeleconsultationLink()" class="dialog-button-group start-tele-consultation"
```
[the remark this answers is not in the document]
Comment (vishalreddysarasani) [reply ui/app/clinical/dashboard/views/clinicalDashboardHeader.html:85]: Yes @gsluthra it defaults to false

File: ui/app/clinical/consultation/controllers/consultationController.js [hunk 2]
```diff
@@ -58,6 +60,31 @@ angular.module('bahmni.clinical').controller('ConsultationController',
...
+                                var message = data.data.notificationResults[0].message;
+                                var status = data.data.notificationResults[0].status;
+                                if (status === 1) {
+                                    messagingService.showMessage('error', message);
+                                } else {
+                                    messagingService.showMessage('info', message);
+                                }
+                            }
+                        } else {
+                            messagingService.showMessage('error', "{{'TELECON_ERROR_KEY' | translate }}");
```
[the remark this answers is not in the document]
Comment (vishalreddysarasani) [reply ui/app/clinical/consultation/controllers/consultationController.js:73]: Changed

File: ui/app/clinical/consultation/controllers/consultationController.js [hunk 3]
```diff
@@ -58,6 +60,31 @@ angular.module('bahmni.clinical').controller('ConsultationController',
...
+                    {
+                        patientUuid: patientContext.patient.uuid,
+                        provider: $rootScope.currentUser.username
+                    }).then(function (data) {
+                        if (data && data.data) {
+                            virtualConsultService.launchMeeting(data.data.uuid, data.data.link);
+                            if (data.data.notificationResults && data.data.notificationResults.length > 0) {
+                                var message = data.data.notificationResults[0].message;
+                                var status = data.data.notificationResults[0].status;
+                                if (status === 1) {
```
[the remark this answers is not in the document]
Comment (vishalreddysarasani) [reply ui/app/clinical/consultation/controllers/consultationController.js]: Using constant

**Review by gsluthra — APPROVED** [review]

**Comment by kushboojain** [comment]
The changes are moved to another PR and its been merged.

**Pull request closed without merging**